packages/components/src/color-palette/types.ts

```diff
@@ -0,0 +1,85 @@
+/**
+ * External dependencies
+ */
+import type { CSSProperties, MouseEventHandler, ReactNode } from 'react';
+
+type OnColorChange = ( newColor?: string ) => void;
+
+export type ColorObject = {
+	name: string;
+	color: NonNullable< CSSProperties[ 'color' ] >;
+};
+
+export type PaletteObject = {
+	name: string;
+	colors: ColorObject[];
+};
+
+type PaletteProps = {
+	className?: string;
+	clearColor: () => void;
+	onChange: OnColorChange;
+	value?: string;
+	actions?: ReactNode;
+};
+
+export type SinglePaletteProps = PaletteProps & {
+	colors: ColorObject[];
+};
+
+export type MultiplePalettesProps = PaletteProps & {
+	colors: PaletteObject[];
+};
+
+export type CustomColorPickerDropdownProps = {
+	isRenderedInSidebar: boolean;
+	renderContent: () => JSX.Element;
+	popoverProps?: string[];
+	renderToggle: ( props: {
+		isOpen: boolean;
+		onToggle: MouseEventHandler< HTMLButtonElement >;
+	} ) => JSX.Element;
+};
+
+export type ColorPaletteProps = {
+	/**
+	 * Whether the palette should have a clearing button.
+	 */
+	clearable?: boolean;
```

Comment:
Looking at the [component's props destructuring](https://github.com/WordPress/gutenberg/blob/6ee6987a2c392b9f154339e3a11d5f9ca0bb22e5/packages/components/src/color-palette/index.tsx#L217-L225), a few props are missing their `@default` type annotation

```diff
diff --git a/packages/components/src/color-palette/types.ts b/packages/components/src/color-palette/types.ts
index 1c8f6ec1a8..a940ff09a0 100644
--- a/packages/components/src/color-palette/types.ts
+++ b/packages/components/src/color-palette/types.ts
@@ -44,6 +44,8 @@ export type CustomColorPickerDropdownProps = {
 export type ColorPaletteProps = {
 	/**
 	 * Whether the palette should have a clearing button.
+	 *
+	 * @default true
 	 */
 	clearable?: boolean;
 	/**
@@ -58,6 +60,8 @@ export type ColorPaletteProps = {
 	colors?: ( PaletteObject | ColorObject )[];
 	/**
 	 * Whether to allow custom color or not.
+	 *
+	 * @default false
 	 */
 	disableCustomColors?: boolean;
 	/**
@@ -76,10 +80,14 @@ export type ColorPaletteProps = {
 	/**
 	 * Whether the colors prop is an array of color palettes,
 	 * rather than an array of color objects.
+	 *
+	 * @default false
 	 */
 	__experimentalHasMultipleOrigins?: boolean;
 	/**
 	 * Whether this is rendered in the sidebar.
+	 *
+	 * @default false
 	 */
 	__experimentalIsRenderedInSidebar?: boolean;
 };
```

Reply:
Thanks, committed in https://github.com/WordPress/gutenberg/pull/44632/commits/9b770d3b07dd216aafb9c566deadd15ea5842794